### ISSUE TITLE
No noImplicitAny error for self referencing values

### ISSUE THREAD

**Issue by mhegazy** [microsoft/TypeScript#476]
This should cause an error with --noImplicitAny

``` TypeScript
var x : typeof x;  // x is typed as any, but no error with --noImplicitAny
```

similarly

``` TypeScript
var x = x;
```
